java-agent/instrumentation/servlet/src/main/java/io/opentelemetry/auto/instrumentation/servlet/http/HttpServletInstrumentation.java

```diff
@@ -67,31 +65,33 @@ public boolean defaultEnabled() {
   public static class HttpServletAdvice {
 
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static AgentScope start(@Advice.Origin final Method method) {
+    public static SpanScopePair start(@Advice.Origin final Method method) {
 
-      if (activeSpan() == null) {
+      if (TRACER.getCurrentSpan() == null || !TRACER.getCurrentSpan().getContext().isValid()) {
```

Comment:
```suggestion
      if (!TRACER.getCurrentSpan().getContext().isValid()) {
```
TRACER.getCurrentSpan() never returns `null`